src/app/backend/apihandler.go

```diff
@@ -297,11 +297,24 @@ func (apiHandler *ApiHandler) handleUpdateReplicasCount(
 }
 
 // Handles delete Replication Controller API call.
+// TODO(floreks): there has to be some kind of transaction here
 func (apiHandler *ApiHandler) handleDeleteReplicationController(
 	request *restful.Request, response *restful.Response) {
 
 	namespace := request.PathParameter("namespace")
 	replicationController := request.PathParameter("replicationController")
+	deleteServices, err := strconv.ParseBool(request.QueryParameter("deleteServices"))
+	if err != nil {
+		handleInternalError(response, err)
+		return
+	}
+
+	if deleteServices {
```

Comment:
Apihandlers should be logic-less. Can you move this to DeleteReplicationController and pass the boolean to it? The idea is to have zero logic in this file and just pass arguments to implementation functions.